src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,256 @@
 
-    public OrdersKafkaConsumer(SerializerFactory serializerFactory) {
+    public OrdersKafkaConsumer(SerializerFactory serializerFactory,
+                               OrdersKafkaProducer kafkaProducer,
+                               KafkaListenerEndpointRegistry registry) {
         this.serializerFactory = serializerFactory;
+        this.kafkaProducer = kafkaProducer;
+        this.registry = registry;
     }
 
-    @Override
-    public void afterPropertiesSet() throws Exception {
-        LOGGER.debug("Initializing kafka consumer service " + this.toString());
-
-        ProducerConfig config = getProducerConfig();
-        CHKafkaProducer chKafkaProducer = new CHKafkaProducer(config);
-
-        ConsumerConfig consumerConfig = getConsumerConfig();
-
-        consumerConfig.setGroupName(ORDER_RECEIVED_GROUP);
-        consumerConfig.setTopics(Collections.singletonList(ORDER_RECEIVED_TOPIC));
-        chKafkaConsumerGroupMain
-                = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.MAIN_CONSUMER,
-                        new KafkaConsumerFactory(), chKafkaProducer);
+    /**
+     * Main listener/consumer. Calls a `retryable` method to process received message.
+     * If the `retryable` processor is unsuccessful with a `retryable` error, after maximum numbers of attempts allowed,
+     * the message is published to `-error` topic for failover processing.
+     * @param message
+     * @throws SerializationException
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    @KafkaListener(id = ORDER_RECEIVED_GROUP, groupId = ORDER_RECEIVED_GROUP,
+                    topics = ORDER_RECEIVED_TOPIC,
+                    autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
+    public void processOrderReceived(org.springframework.messaging.Message<OrderReceived> message)
+            throws SerializationException, ExecutionException, InterruptedException {
+        OrderReceived orderReceived = message.getPayload();
+        String orderReceivedUri = orderReceived.getOrderUri();
+        try {
+            logMessageReceived(message, ORDER_RECEIVED_TOPIC);
 
-        consumerConfig.setTopics(Collections.singletonList(ORDER_RECEIVED_TOPIC_RETRY));
-        consumerConfig.setGroupName(ORDER_RECEIVED_GROUP_RETRY);
-        chKafkaConsumerGroupRetry
-                = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.RETRY_CONSUMER,
-                        new KafkaConsumerFactory(), chKafkaProducer);
+            logMessageProcessed(message, ORDER_RECEIVED_TOPIC);
+        } catch (ServiceException ex){
+            logMessageProcessingFailureRecoverable(message, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_ERROR, ex);
+            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_ERROR);
+        } catch (Exception x) {
+            logMessageProcessingFailureNonRecoverable(message, ORDER_RECEIVED_TOPIC, x);
```

Comment:
Are all exceptions definitely not recoverable? If a bug in this app causes an exception we can fix the issue and re-process the messages.

Reply:
The recoverable errors are thrown/caught using ServiceException (which the proposed EmailService will implement). The email service will use Spring's Retryable feature to offer a method that will retry all recoverable errors at least 3 times before propagating the error here. At that point the unhandled/unprocessed message will be republished to `-error` topic. The non-recoverable errors are logged and there is no further processing.

Reply:
If all retryable exceptions are being caught by ServiceException, what's being caught by Exception? We shouldn't be catching generic exceptions (unless Spring is throwing an Exception) because it's unclear reading the code what that catch block is catching.

The email service shouldn't be handling the retry logic. Retrying logic should be in a reusable place (e.g. Kafka library). Where's the design for this?

Reply:
Since this story doesn't implement the actual processing of the message, the exceptions are currently just placeholders. This exception handling will change when integrated with the EmailService.

Reply:
Can it be removed rather than a placeholder please? This could be left unmodified if someone else adds the processing without realising it needed updating. That would risk bugs being introduced.

Reply:
Sure. Makes sense.

Reply:
The logic to republish to next topic (when retries are exhausted) will be implemented in the message processing story.